File: closest_two_sum_to_zero.swift
// http://programmingpraxis.com/2015/06/23/closest-two-sum-to-zero/

func closestTwoSumToZero(array: [Int]) -> [Int] {
  var closestElementOne = array[0]
  var closestElementTwo = array[1]
  var distanceFromZero = abs(array[0] + array[1])
  for (var x = 1; x < array.count; x++) {
    for (var y = x+1; y < array.count; y++) {
      if (abs(array[x] + array[y]) < distanceFromZero) {
        closestElementOne = array[x]
        closestElementTwo = array[y]
        distanceFromZero = abs(array[x] + array[y])
      }
    }
  }
  return [closestElementOne,closestElementTwo]
}

let test = [45, -29, -96, -7, -17, 72, -60]
println(closestTwoSumToZero(test))

File: closest_two_sum_to_zero.rb
#!/usr/bin/env ruby
# http://programmingpraxis.com/2015/06/23/closest-two-sum-to-zero/

def closest_two_sum_to_zero(array)
  closest_element_one = array[0] 
  closest_element_two = array[1]
  distance_from_zero = (array[0] + array[1]).abs
  (1...array.count).each do |x|
    (x+1...array.count).each do |y|
      if (array[x] + array[y]).abs < distance_from_zero
        closest_element_one = array[x]
        closest_element_two = array[y]
        distance_from_zero = (array[x] + array[y]).abs
      end
    end
  end
  [closest_element_one,closest_element_two]
end

test = [45, -29, -96, -7, -17, 72, -60]
p closest_two_sum_to_zero(test)
